src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py

```diff
@@ -588,139 +588,114 @@ def test_graph_user_scenario(self):
+                checks=[self.check('displayName', '{group}'),
+                        self.check('mailNickname', '{mail_nick_name}'),
+                        self.check('description', '{group}')]
+            ).get_output_in_json()
             self.kwargs['group_id'] = group_result['id']
-            # add user1 into group
-            self.cmd('ad group member add -g {group} --member-id {user1_id}',
-                     checks=self.is_empty())
-            # add user2 into group
-            self.cmd('ad group member add -g {group} --member-id {user2_id}',
-                     checks=self.is_empty())
-
-            # show user's group memberships
-            self.cmd('ad user get-member-groups --upn-or-object-id {user1_id}',
-                     checks=self.check('[0].displayName', self.kwargs['group']))
+
+            # create again to test idempotency
+            self.cmd('ad group create --display-name {group} --mail-nickname {mail_nick_name}')
+            # list groups
+            self.cmd('ad group list --display-name {group}', checks=self.check('length([])', 1))
             # show group
             self.cmd('ad group show -g {group}', checks=[
                 self.check('id', '{group_id}'),
                 self.check('displayName', '{group}'),
+                self.check('mailNickname', '{mail_nick_name}'),
                 self.check('description', '{group}')
             ])
-            self.cmd('ad group show -g {group}',
-                     checks=self.check('displayName', '{group}'))
-            # list group
-            self.cmd('ad group list --display-name {group}',
-                     checks=self.check('[0].displayName', '{group}'))
-            # show member groups
-            self.cmd('ad group get-member-groups -g {group}',
-                     checks=self.check('length([])', 0))
-            # check user1 memebership
-            self.cmd('ad group member check -g {group} --member-id {user1_id}',
-                     checks=self.check('value', True))
 
-            # check user2 memebership
-            self.cmd('ad group member check -g {group} --member-id {user2_id}',
+            # create other groups to test membership transitivity
+            group_result = self.cmd('ad group create --display-name {child_group} --mail-nickname {mail_nick_name}').get_output_in_json()
+            self.kwargs['child_group_id'] = group_result['id']
+            group_result = self.cmd('ad group create --display-name {leaf_group} --mail-nickname {mail_nick_name}').get_output_in_json()
+            self.kwargs['leaf_group_id'] = group_result['id']
+
+            # add child_group as member of group
+            self.cmd('ad group member add -g {group_id} --member-id {child_group_id}')
+            # add leaf_group as member of child_group
+            self.cmd('ad group member add -g {child_group_id} --member-id {leaf_group_id}')
```

Comment:
Let's make sure we also test adding users as members.

Reply:
Added test for user and service principal